src/main.rs

```diff
@@ -1,3 +1,225 @@
-fn main() {
-    println!("hello, world!");
+use clap::{Args, Parser, Subcommand};
+use felt::Felt;
+use num_traits::{Num, Zero};
+use starknet_rs::{
+    business_logic::{
+        execution::{
+            execution_entry_point::ExecutionEntryPoint,
+            objects::{CallType, TransactionExecutionContext},
+        },
+        fact_state::{
+            contract_state::ContractState, in_memory_state_reader::InMemoryStateReader,
+            state::ExecutionResourcesManager,
+        },
+        state::cached_state::CachedState,
+    },
+    core::{
+        contract_address::starknet_contract_address::compute_class_hash,
+        transaction_hash::starknet_transaction_hash::{
+            calculate_declare_transaction_hash, calculate_deploy_transaction_hash,
+            calculate_transaction_hash_common, TransactionHashPrefix,
+        },
+    },
+    definitions::{constants::TRANSACTION_VERSION, general_config::StarknetGeneralConfig},
+    hash_utils::calculate_contract_address,
+    parser_errors::ParserError,
+    serde_structs::contract_abi::read_abi,
+    server::{
+        add_address, add_class_hash, get_contract_from_address, get_contract_from_class_hash,
+    },
+    services::api::contract_class::ContractClass,
+    utils::{felt_to_hash, Address},
+};
+use std::{collections::HashMap, path::PathBuf};
+
+#[derive(Parser)]
+struct Cli {
+    #[command(subcommand)]
+    command: Commands,
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    Declare(DeclareArgs),
+    Deploy(DeployArgs),
+    Invoke(InvokeArgs),
+}
+
+#[derive(Args)]
+struct DeclareArgs {
+    #[arg(long)]
+    contract: PathBuf,
+}
+
+#[derive(Args)]
+struct DeployArgs {
+    #[arg(long = "class_hash")]
+    class_hash: String,
+    #[arg(long, default_value = "1111")]
+    salt: i32,
+    #[arg(long, num_args=1.., value_delimiter = ' ')]
+    inputs: Option<Vec<i32>>,
+}
+
+#[derive(Args)]
+struct InvokeArgs {
+    #[arg(long)]
+    address: String,
+    #[arg(long)]
+    abi: PathBuf,
+    #[arg(long)]
+    function: String,
+    #[arg(long, num_args=1.., value_delimiter = ' ')]
+    inputs: Option<Vec<i32>>,
+}
+
+fn declare_parser(args: &DeclareArgs) -> Result<(), ParserError> {
+    let contract_class =
+        ContractClass::try_from(&args.contract).map_err(ParserError::OpenFileError)?;
+    let class_hash =
+        compute_class_hash(&contract_class).map_err(ParserError::ComputeClassHashError)?;
+    add_class_hash(&(&class_hash).to_biguint().to_str_radix(16), &args.contract);
+
+    let tx_hash = calculate_declare_transaction_hash(
+        contract_class,
+        Felt::zero(),
+        Address(0.into()),
+        0,
+        0,
+        0.into(),
```

Comment:
```suggestion
        Address(Felt::zero()),
        0,
        0,
        Felt::zero(),
```
To keep consistency it is better to use `Felt::zero()` that using `into()`